TheMovieDb/Modules/Home/HomeViewPresenter.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  HomeViewPresenter.swift
+//  TheMovieDb
+//
+//  Created by Ricardo Ramirez on 20/11/21.
+//
+
+import Foundation
+import OSLog
+
+protocol HomeViewPresenterDelegate: AnyObject {
+    func didStartLoading()
+    func didFinishLoading()
+    func didStartSearching()
+    func didFinishSearching()
+    func didUpdateMovies(_ movies: [Movie])
+}
+
+class HomeViewPresenter {
+    
+    let service: MovieFeedRepository
+    
+    weak var delegate: HomeViewPresenterDelegate?
+    
+    var movies = [Movie]()
+    
+    var isLoading = false {
+        didSet {
+            isLoading ? delegate?.didStartLoading() : delegate?.didFinishLoading()
+        }
+    }
+    
+    var loadedPages: Int = .zero
+    
+    var isSearching = false {
+        didSet {
+            currentFeed = isSearching ? .search : .trending
+            isSearching ? delegate?.didStartSearching() : delegate?.didFinishSearching()
+        }
+    }
+    
+    var currentFeed: FeedTypes = .trending {
+        didSet {
+            resetFeed()
+            if !isSearching {
+                getMoviesIfNeeded(search: nil)
+            }
+        }
+    }
```

Comment:
The view controller can't be allowed to access the Models

Reply:
Solved